src/EditorFeatures/CSharpTest/MisplacedUsings/MisplacedUsingsCodeFixProviderTests.cs

```diff
@@ -0,0 +1,1295 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.MisplacedUsings;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MisplacedUsings
+{
+    using Verify = CSharpCodeFixVerifier<MisplacedUsingsDiagnosticAnalyzer, MisplacedUsingsCodeFixProvider, XUnitVerifier>;
+
+    /// <summary>
+    /// Unit tests for the <see cref="MisplacedUsingsDiagnosticAnalyzer"/> and <see cref="MisplacedUsingsCodeFixProvider"/>.
+    /// </summary>
+    public class MisplacedUsingsCodeFixProviderTests
```

Comment:
i'm confused why this isn't just using our normal code fix base class.

Reply:
Most of the deviations on this front are the result of avoiding unnecessary deviations from the original implementation in the initial submission. There was no "normal code fix base class" in the original.

Reply:
I'm ok with that as a first draft.  But it does'nt make sense to me in terms of merging in.  Will we just be producing more and more duplication of code that may already exist just because it's in StyleCop analyzers?  I def do not like that. This just means more code to maintain and fixup when we encounter bugs and whatnot.

Reply:
Roslyn is one of the remaining outlies that does not use Microsoft.CodeAnalysis.Testing, the new common library for analyzer and code fix testing. The default configuration for that library includes a large number of much stronger assertions regarding "well-behaved" implementations, so the long term goal is migrating all current testing to be built on that library.